pkg/scheduler/framework/plugins/dynamicresources/dynamicresources_test.go

```diff
@@ -197,6 +198,74 @@ type prepare struct {
 	postfilter change
 }
 
+func BenchmarkHaveAllNodes(b *testing.B) {
+	for nodeCount := 10; nodeCount < 200; nodeCount += 10 {
+		var nodeNames []string
+		var nodes []*v1.Node
+		for i := nodeCount; i > 0; i-- {
+			nodeNames = append(nodeNames, "worker-"+strconv.Itoa(i))
+			nodes = append(nodes, &st.MakeNode().Name("worker-"+strconv.Itoa(i)).Node)
+		}
+		b.ResetTimer()
+		b.Run(fmt.Sprintf("have-nodes-%d", nodeCount), func(b *testing.B) {
+			for i := 0; i < b.N; i++ {
+				haveAllNodes(nodeNames, nodes)
+			}
+		})
+	}
+}
+
+func TestHaveAllNodes(t *testing.T) {
+	tests := []*struct {
+		name      string
+		nodeNames []string
+		nodes     []*v1.Node
+		has       bool
+	}{
+		{
+			name:      "same-nodes",
+			nodeNames: []string{"worker-a"},
+			nodes: []*v1.Node{
+				&st.MakeNode().Name("worker-a").Node,
+			},
+			has: true,
+		},
+		{
+			name:      "have-all",
+			nodeNames: []string{"worker-a", "worker-b"},
+			nodes: []*v1.Node{
+				&st.MakeNode().Name("worker-a").Node,
+			},
+			has: true,
+		},
+		{
+			name:      "not-all",
+			nodeNames: []string{"worker-a"},
+			nodes: []*v1.Node{
+				&st.MakeNode().Name("worker-a").Node,
+				&st.MakeNode().Name("worker-b").Node,
+			},
+			has: false,
+		},
+	}
```

Comment:
Please, add test cases covering `if len(nodeNames) < len(nodes) {` code and empty slices, thanks.